ports/psoc6/modules/psoc6/psoc6_fatfs.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2013, 2014 Damien P. George
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <stdio.h>
+
+
+#include "lib/oofatfs/ff.h"
+
+
+#include "cyhal.h"
+
+
+// object defined in main
+extern cyhal_rtc_t psoc6_rtc;
+
+
+MP_WEAK DWORD get_fattime(void) {
+    struct tm current_date_time = {0};
+    cy_rslt_t result = cyhal_rtc_read(&psoc6_rtc, &current_date_time);
+
+    if (CY_RSLT_SUCCESS != result) {
+        // mp_raise_NotImplementedError("cyhal_rtc_read failed !");
+        printf("cyhal_rtc_read failed !");
```

Comment:
As an alternative to a simple printf MPY provides also a function to raise custom exceptions like here: 
https://github.com/jaenrig-ifx/micropython/blob/82a146a2f02be2ba67c5698d4a720dbc8d9d4959/ports/psoc6/modules/machine/machine_pin.c#L196

Reply:
I changed it to what it have should been : mp_raise_NotImplementedError(MP_ERROR_TEXT(...))